src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PluginSettings.java

```diff
@@ -271,4 +295,21 @@ private void loadMetricsDBFilesCleanupEnabled() {
       shouldCleanupMetricsDBFiles = true;
     }
   }
+
+  private void loadBatchMetricsRetentionPeriodMinutesFromConfig() {
+    if (!settings.containsKey(BATCH_METRICS_RETENTION_PERIOD_MINUTES)) {
+      return;
+    }
+
+    try {
+      long parsedRetentionPeriod = Long.parseLong(settings.getProperty(BATCH_METRICS_RETENTION_PERIOD_MINUTES));
+      if (parsedRetentionPeriod <= 0 || parsedRetentionPeriod > BATCH_METRICS_RETENTION_PERIOD_MINUTES_LIMIT) {
+        throw new InvalidParameterException();
+      }
+      batchMetricsRetentionPeriodMinutes = parsedRetentionPeriod;
+    } catch (NumberFormatException | InvalidParameterException e) {
+      LOG.error("Invalid batch-metrics-retention-period-minutes. Using default value {}.",
+              batchMetricsRetentionPeriodMinutes, e);
```

Comment:
Do we need the stack trace here ? 
And as this is logged for both `NumberFormatException` and `InvalidParameterException`, it might make sense to print `(settings.getProperty(BATCH_METRICS_RETENTION_PERIOD_MINUTES)` to understand which of the two it was by just looking at the logs ?

Reply:
Sure will get rid of the exception log and doing something like this.